pinc/graph_data.inc

```diff
@@ -650,64 +650,45 @@ function pages_daily($tally_name, $c_or_i, $timeframe)
     if ($c_or_i == 'increments') {
         $days_to_average = 21;
 
-        // to ensure we have enough data to use, go back $days_to_average
-        // days before the start date
-        $where_start_timestamp = $start_timestamp - ($days_to_average * 60 * 60 * 24);
-
-        // Unless it's for all_time when start_timestamp==0. For this case we need
-        // $days_to_average days after the timestamp is zero to ensure the average
-        // is correct.
-        if ($start_timestamp == 0) {
-            $where_start_timestamp = $start_timestamp + ($days_to_average * 60 * 60 * 24);
+        // To calculate a simple moving average over $days_to_average we need
+        // to have at least that many days worth of data before $start_timestamp.
+        // This is simple for everything except right at the beginning when
+        // stats were collected. For that we need to jump through some hoops
+        // to 1) detect it and 2) pad the data with 0s so that we can calculate
```

Comment:
It seems to me we might want to pad with the first value, not with 0s... That way, the SMA gets closer and closer to the first value the closer `$sma_start_timestamp` gets to `$start_timestamp`

Reply:
But then we skew the SMAs for all the subsequent counts over the next 21 days because there weren't $first_value pages proofread before we started collecting stats. 0 isn't right, but $first_value isn't either (and 0 is easier). The "right" think is probably to not an SMA for the first 21 days, or to have a sliding window for the first 0 to 21 days.

Reply:
> The "right" think is probably to not an SMA for the first 21 days, or to have a sliding window for the first 0 to 21 days.

That's what makes the most sense to me.

Reply:
Right, padding with first-value or with zero gives unwarranted weight to that value until the padding falls out of the window.

> or to have a sliding window for the first 0 to 21 days

(Well, we use a sliding window for *all* days.) I'm guessing what you mean is just average whatever real data exists in the window. (E.g., to get the SMA for day 5, just take the average of days 1 through 5.) That'd be my choice.